pkg/coremodel/pluginmeta/coremodel.cue

```diff
@@ -0,0 +1,340 @@
+				executable?: string
+
+				// Initialize plugin on startup. By default, the plugin
+				// initializes on first use.
+				preload?: bool
+
+				// Marks a plugin as a pre-release.
+				state?: #ReleaseState
+
+				// ReleaseState indicates release maturity state of a plugin.
+				#ReleaseState: "alpha" | "beta" | *"stable"
+
+				// Resources to include in plugin.
+				includes?: [...#Include]
+
+				// A resource to be included in a plugin.
+				#Include: {
+					// Unique identifier of the included resource
+					uid?:  string
+					type:  #IncludeType
+					name?: string
+
+					// (Legacy) The Angular component to use for a page.
+					component?: string
+					role?:      "Admin" | "Editor" | "Viewer"
+
+					// Used for app plugins.
+					path?: string
+
+					// Add the include to the side menu.
+					addToNav?: bool
+
+					// Page or dashboard when user clicks the icon in the side menu.
+					defaultNav?: bool
+
+					// Icon to use in the side menu. For information on available
+					// icon, refer to [Icons
+					// Overview](https://developers.grafana.com/ui/latest/index.html?path=/story/docs-overview-icon--icons-overview).
+					icon?: string
+					...
+				}
+
+				// For data source plugins, if the plugin supports logs.
+				logs?: bool
+
+				// For panel plugins. Hides the query editor.
+				skipDataQuery?: bool
+
+				// For data source plugins, if the plugin supports metric queries.
+				// Used in Explore.
+				metrics?: bool
+
+				// For data source plugins, if the plugin supports streaming.
+				streaming?: bool
+
+				// This is an undocumented feature.
+				tables?: bool
+
+				// For data source plugins, if the plugin supports tracing.
+				tracing?: bool
+
+				// For data source plugins, include hidden queries in the data
+				// request.
+				hiddenQueries?: bool
+
+				// Set to true for app plugins that should be enabled by default
+				// in all orgs
+				autoEnabled?: bool
+
+				// Dependencies needed by the plugin.
+				dependencies: #Dependencies
+
+				#Dependencies: {
+					// (Deprecated) Required Grafana version for this plugin, e.g.
+					// `6.x.x 7.x.x` to denote plugin requires Grafana v6.x.x or
+					// v7.x.x.
+					grafanaVersion?: =~"^([0-9]+)(\\.[0-9x]+)?(\\.[0-9x])?$"
+
+					// Required Grafana version for this plugin. Validated using
+					// https://github.com/npm/node-semver.
+					grafanaDependency: =~"^(<=|>=|<|>|=|~|\\^)?([0-9]+)(\\.[0-9x\\*]+)(\\.[0-9x\\*])?(\\s(<=|>=|<|=>)?([0-9]+)(\\.[0-9x]+)(\\.[0-9x]))?$"
+
+					// An array of required plugins on which this plugin depends.
+					plugins?: [...#Dependency]
+				}
+
+				// Dependency describes another plugin on which a plugin depends.
+				// The id refers to the plugin package identifier, as given on
+				// the grafana.com plugin marketplace.
+				#Dependency: {
+					id:      =~"^[0-9a-z]+\\-([0-9a-z]+\\-)?(app|panel|datasource)$"
+					type:    "app" | "datasource" | "panel"
+					name:    string
```

Comment:
Not even by external plugins?

Reply:
Maybe they specify it, but Grafana never uses it 🤷

Reply:
yeah but if they do that, and we don't allow it - at least in the initial version of the schema - then it will make all those existing files invalid.

Gotta keep it for 0.0 - can remove it in 1.0

Reply:
Hah! Gotta get used to this whole "schema" thing 😅